### PR TITLE
Reduce overhead by fixing alignment in PerPriority

### PR DESCRIPTION
**Pull request by DiazAlan**
Commit Message: Reduce overhead by fixing alignment in PerPriority
Additional Description: Moves many of the smaller members of the class to the end to save small amounts of memory
Risk Level: Low

### CODE REVIEW

File: source/common/upstream/cluster_manager_impl.h
Comment (adisuissa) [inline source/common/upstream/cluster_manager_impl.h:380]: nit: please add a small note about the coalescing of these data members at the end (something like [this](https://github.com/envoyproxy/envoy/blob/1ac0a5948ca317fd418777dab2ffe27935c0a276/source/common/router/config_impl.h#L342)).

**Review by adisuissa — COMMENTED** [review]
Thanks!
LGTM modulo nit.

File: source/common/upstream/cluster_manager_impl.h
Comment (DiazAlan) [reply source/common/upstream/cluster_manager_impl.h:381]: Done

**Review by adisuissa — APPROVED** [review]
LGTM, thanks!

/assign-from @envoyproxy/senior-maintainers

**Review by zuercher — APPROVED** [review]
Thanks!

**Pull request merged**
